sdk/eventhubs/azure_messaging_eventhubs/examples/eventhubs_batch_produce_events.rs

```diff
@@ -0,0 +1,61 @@
+/// This sample demonstrates how to send events to all partitions using a batch sender.
+///
+use azure_identity::DefaultAzureCredential;
+use azure_messaging_eventhubs::{models::EventData, EventDataBatchOptions, ProducerClient};
+use uuid::Uuid;
+
+#[tokio::main]
+async fn main() -> Result<(), Box<dyn std::error::Error>> {
+    // Set up the Event Hub client
+    let eventhub_namespace = std::env::var("EVENTHUBS_HOST")?;
+    let eventhub_name = std::env::var("EVENTHUB_NAME")?;
+    let credential = DefaultAzureCredential::new()?;
+
+    let client = ProducerClient::new(eventhub_namespace, eventhub_name, credential, None);
```

Comment:
Even if it's the only one, we want to show best practices by cloning the arc:
```suggestion
    let client = ProducerClient::new(eventhub_namespace, eventhub_name, credential.clone(), None);
```